### ISSUE TITLE
Location of connections informations and queries executions history

### ISSUE THREAD

**Issue by FALLAI-Denis** [IBM/db2forzosdeveloperextension-about#23]
On a Windows VS Code platform, information regarding connections to Db2 z/OS systems and the history of query execution is placed outside of the Workspace (which may be a Git repository), or the configuration of VS Code.
This location is `%APPDATA%\Code\User\globalStorage\ibm.db2forzosdeveloperextension`.

We believe that some information should instead be stored either in the VS Code configuration, user's "settings.json" file, or in the Workspace in use.

`connections.json`: configuration of a connection to the z/OS Db2 servers, including part of credentials (userid).
Some informations are shareable by all users (location of the z/OS server), and could appear in the "settings.json" file in the Workspace, or a dedicated file in the Workspace.
Other information is attached to a user (identification information) and could appear in the User's "settings.json" file.

`editor_connections.json`: association between an SQL source file and a connection to a Db2 z/OS system.
This information can be shared by all Users and could appear in the "settings.json" file in the Workspace or a dedicated file in the Workspace.

`sql_execution\history.json`: history of all executions.
This information is linked to a User. They are volatile and should not be stored for an indefinite period of time.
They could be managed in a temporary folder, or at least one cleanup is planned.

`sql_execution\file_ext\timestamp.json`: result of an execution, including the recovered data (security risk).
This information is a priori linked to a User, (but could also be shared by all Users, if they were not dependent on the `history.json` file). They are volatile and should not be stored for an indefinite period of time.
They could be managed in a temporary folder, or at least one cleanup is planned.
Note that the data retrieved from the z/OS server could be sensitive and their conservation can pose problems (bank production data for example).

**Comment by realkendrickren** [comment]
@FALLAI-Denis thanks for the feedback. 

Security and sensitive data were considered on where to put those json files. %APPDATA% (on Windows) is under the user's home directory, which is only accessible to the current user, instead of all users, unless you are the administrator of the workstation or you, as the owner or admin, grant others privilege to access the folder. Query history and result data may contain sensitive data as you also mentioned, so it is not designed to be shared with others by default, if users decide to share the data with others, they can save the data into .csv file and share the .csv file instead.
System temporary folders usually are public to all users, which would potentially expose the sensitive data if there is, even for a short period, so, the extension avoids to put these types of data there.
Regarding how to purge the data, users can choose to clean up specific history/result or entire history/result from Query History view in the extension today.

The association between editor and connection involves the specific directory and file names, which may likely vary among different users and workstation, hence, the association on one user's environment may not apply to others. This information is managed by the extension per user, and the extension purges the information if you no longer have the previous connection.

As for sharing the connection definition like host/location/port with other users, that is a good one, we also have that in our list for future, so I mark this as an enhancement request.

**Comment by Marty850** [comment]
Vijaya asked me to place our user feedback here ;-)

A function to import and export connection
definitions would be great. We have a lot of DB2 subsystems an so it would
be easier for my colleagues to set up the DB2 Developer Extension.

Kind regards
Martin

**Comment by VijayaKatikireddy** [comment]
Thanks Martin!

**Comment by katsoohoo** [comment]
Hi all,

For sharing connection definitions, we recommended sharing the connection definition files. Since the definitions are stored in JSON format, importing/exporting can be done by copying and sharing these files.

We added the [Sharing connection definitions](https://ibm.github.io/db2forzosdeveloperextension-about/docs/tips-and-tricks/sharing-connection-definitions) article to our documentation explaining the how to do that.
